packages/typespec-client-generator-core/test/decorators/deserialize-empty-string-as-null.test.ts

```diff
@@ -0,0 +1,189 @@
+import { expectDiagnostics } from "@typespec/compiler/testing";
+import { deepStrictEqual } from "assert";
+import { beforeEach, describe, it } from "vitest";
+import { SdkTestRunner, createSdkTestRunner } from "../test-host.js";
+
+let runner: SdkTestRunner;
+
+beforeEach(async () => {
+  runner = await createSdkTestRunner({ emitterName: "@azure-tools/typespec-python" });
+});
+
+describe("deserialized empty string as null", () => {
+  it("Apply the decorator to model properties of type 'string' and a Scalar type derived from 'string'", async function () {
+    runner = await createSdkTestRunner(
+      {},
+      { additionalDecorators: ["Azure\\.ClientGenerator\\.Core\\.@deserializeEmptyStringAsNull"] },
+    );
+
+    await runner.compileWithBuiltInService(`
+        scalar stringlike extends string;
+
+        model A {
+          @deserializeEmptyStringAsNull("csharp")
+          prop1: stringlike;
+
+          @deserializeEmptyStringAsNull("csharp")
+          prop2: string;
+        }
+
+        op test(): A;
+      `);
+
+    const models = runner.context.sdkPackage.models;
+
+    for (const prop of [models[0].properties[0], models[0].properties[1]]) {
+      deepStrictEqual(prop.decorators, [
+        {
+          name: "Azure.ClientGenerator.Core.@deserializeEmptyStringAsNull",
+          arguments: { scope: "csharp" },
+        },
+      ]);
+    }
+    expectDiagnostics(runner.context.diagnostics, []);
+  });
+
+  it("Apply decorator to model properties of indirectly derived from 'string'", async function () {
+    runner = await createSdkTestRunner(
+      {},
+      { additionalDecorators: ["Azure\\.ClientGenerator\\.Core\\.@deserializeEmptyStringAsNull"] },
+    );
+
+    await runner.compileWithBuiltInService(`
+        scalar l1 extends string;
+        scalar l2 extends l1;
+
+        model A {
+          @deserializeEmptyStringAsNull("csharp")
```

Comment:
What if there is no scope specified like in spector case defined? Will this misleading customers if some languages don't support this? 

https://github.com/Azure/typespec-azure/blob/main/packages/azure-http-specs/specs/azure/client-generator-core/deserialize-empty-string-as-null/main.tsp#L18-L21

```
@doc("This is a Model contains a string-like property of type url.")
model ResponseModel {
  @deserializeEmptyStringAsNull
  sampleUrl: url;
}
```